### PR TITLE
feat: support tags for postPolicy Upload

### PR DESCRIPTION
**Pull request by jiuker**
feat: support tags for postPolicy Upload
related: https://github.com/minio/minio/pull/19816

### CODE REVIEW

**Review by shtripat — APPROVED** [review]
lgtm

File: post-policy.go [hunk 1]
```diff
@@ -152,6 +152,23 @@ func (p *PostPolicy) SetCondition(matchType, condition, value string) error {
 	return errInvalidArgument("Invalid condition in policy")
 }
 
+// SetTagging - Sets tagging for the object for this policy based upload.
+func (p *PostPolicy) SetTagging(tagging string) error {
+	if strings.TrimSpace(tagging) == "" || tagging == "" {
```
Comment (vadmeste) [inline post-policy.go:157]: `|| tagging == ""` seems to be redundant

**Review by vadmeste — COMMENTED** [review]
I am not too much into adding a new API since we can have multiple operators here, other than `eq`

The user can use SetCondition() I believe (after fixing it)

File: post-policy.go (hunk 1 above)
Comment (ramondeklein) [reply post-policy.go:157]: I see that it's done more often in this file. I'm not a big fan of checking these trimmed values. It's not very likely that clients will issue a space or a newline, so the check is almost never necessary. And if they do specify it, then the check will fail on the back-end. I would prefer to remove all `string.TrimSpace(..)` in this file.

**Review by ramondeklein — APPROVED** [review]

**Comment by jiuker** [comment]
> I am not too much into adding a new API since we can have multiple operators here, other than `eq`
> 
> The user can use SetCondition() I believe (after fixing it)

`SetCondition` will not support `tagging`. And according to the minio pr @ramondeklein mention the document introduced `tagging` is a single special field. So I want to add that into a function @vadmeste

File: post-policy.go (hunk 1 above)
Comment (jiuker) [reply post-policy.go:157]: I have no idea. @harshavardhana @klauspost  any advice?

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**